### PR TITLE
Optimize Clone.sol  yul

### PR DESCRIPTION
**Pull request by RenanSouza2**
This PR optmizes the yul in Clones.sol

This optmization can be done because solidity already verify dirty bytes in addresses from calldata

There are other optmizations that can be done but those woud require th use of unallocated memory area

Metrics (with `npm run gas-report`):
```
·--------------------------------------|--------|--------|------|---------·
| Method                               · Before · After  · Diff · % Diff  |
·······································|········|········|······|·········|
| Deployments                          · 271943 · 270215 · 1296 ·  0.635% |
·······································|········|········|······|·········|
| $clone(address)                      ·  64079 ·  64070 ·    9 ·  0.014% |
·······································|········|········|······|·········|
| $cloneDeterministic(address,bytes32) ·  64654 ·  64646 ·    8 .  0.012% |
·--------------------------------------|--------|--------|------|---------·
```

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [ ] Changeset entry (run `npx changeset add`)

### CODE REVIEW

**Comment by ernestognw** [comment]
Optimized the title lmao

File: contracts/proxy/Clones.sol [hunk 1]
```diff
@@ -4,7 +4,7 @@
 pragma solidity ^0.8.20;
 
 /**
- * @dev https://eips.ethereum.org/EIPS/eip-1167[ERC-1167] is a standard for
+ * @dev https://eips.ethereum.org/EIPS/eip-1167 [ERC-1167] is a standard for
```
Comment (Amxx) [inline contracts/proxy/Clones.sol:7]: I think the space here is an typo
```suggestion
 * @dev https://eips.ethereum.org/EIPS/eip-1167[ERC-1167] is a standard for
```

Comment (RenanSouza2) [reply contracts/proxy/Clones.sol:7]: It is to fix the link

Comment (ernestognw) [reply contracts/proxy/Clones.sol:7]: > It is to fix the link

Do you mean in VS Code? This is how it works for the documentation in AsciiDoc, using [URL Macros](https://docs.asciidoctor.org/asciidoc/latest/macros/url-macro/)

**Review by ernestognw — COMMENTED** [review]
Looks good to me. We need to fix the NatSpec and also get a second approval from @Amxx

File: contracts/proxy/Clones.sol (hunk 1 above)
Comment (RenanSouza2) [reply contracts/proxy/Clones.sol:7]: ohhh, majes sense

**Review by Amxx — APPROVED** [review]

**Comment by Amxx** [comment]
For posterity, I tried:

```solidity
/// Alternative 1
mstore(0x00, shl(0x60, 0x3d602d80600a3d3981f3363d3d373d3d3d363d73))
mstore(0x14, shl(0x60, implementation))
mstore(0x28, shl(0x88, 0x5af43d82803e903d91602b57fd5bf3))
instance := create(0, 0x00, 0x37)

/// Alternative 2
mstore(0x00, 0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000)
mstore(0x14, shl(0x60, implementation))
mstore(0x28, 0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000)
instance := create(0, 0x00, 0x37)
```

Using 0.8.24 I get:
- Alternative 1 is always 6 gas more expensive than this PR
- Alternative 2 is also 6 gas more expensive if optimisations are enabled, but is 6 gas cheaper when optimisations are turned off.
(Its difficult to isolate the memory preparation from the rest of the function, so devs may want to re-check my numbers with more tooling)

**Comment by Amxx** [comment]
We are talking about 0.01% anyway

**Comment by ernestognw** [comment]
@gitpoap-bot @RenanSouza2

**Pull request merged**